src/states/elder/mod.rs

```diff
@@ -714,24 +714,30 @@ impl Elder {
             msg.routing_message()
         );
 
-        let dst = msg.message_dst();
-        let signed_msg = msg.signed_routing_message();
-        if self.in_location(dst) {
-            self.check_signed_message_trust(signed_msg)?;
-            self.check_signed_message_integrity(signed_msg)?;
-            self.update_our_knowledge(signed_msg);
-
-            if dst.is_multiple() {
-                // Broadcast to the rest of the section.
-                if let Err(error) = self.send_signed_message(&msg) {
-                    debug!("{} Failed to send {:?}: {:?}", self, signed_msg, error);
-                }
+        if msg.message_dst().is_multiple() {
+            // Relay closer to the destination or broadcast to the rest of our section.
+            if let Err(error) = self.send_signed_message(&msg) {
+                debug!(
+                    "{} Failed to send {:?}: {:?}",
+                    self,
+                    msg.signed_routing_message(),
+                    error
+                );
+            }
+        }
+
+        if self.in_location(msg.message_dst()) {
+            let signed_msg = msg.signed_routing_message();
+            let handle = self.verify_signed_message(signed_msg).map_err(|error| {
```

Comment:
do we actually need the `handle`? i.e can we simplify this function and `verify_signed_message` to return Error or nothing? would that make a change in execution trace?

Reply:
Hm, with the relaying now happening first I think so. Let me give it a try.

Reply:
Or maybe not. We don't want to `log_or_panic` if the proof is too new, but do want to in the other cases.

Reply:
Yes, but that happens from within `verify_signed_message` and does not need to leak out.

Reply:
No, it doesn't anymore. I took that out to be able to use the function in both `handle_filtered_signed_message` and `handle_backlogged_filtered_signed_message`.